src/map/lua/lua_baseentity.cpp

```diff
@@ -8692,7 +8674,17 @@ inline int32 CLuaBaseEntity::hasSpellList(lua_State* L)
     DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
     DSP_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_MOB && m_PBaseEntity->objtype != TYPE_PET);
 
-    lua_pushboolean(L, ((CMobEntity*)m_PBaseEntity)->SpellContainer->HasSpells());
+    lua_pushboolean(L, (static_cast<CMobEntity *>(m_PBaseEntity))->SpellContainer->HasSpells());
+
+    return 1;
+}
+
+inline int32 CLuaBaseEntity::hasPreventActionEffect(lua_State* L)
+{
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_NPC);
```

Comment:
Wait.  This code should only take effect on NPCs.  Pets, Mobs, etc.  Therefore we should DEBUG_BREAK_IF type != NPC.  To use == TYPE_NPC would make this code only run on PCs.  Where am I confused.